src/axom/inlet/docs/sphinx/advanced_types.rst

```diff
@@ -65,51 +115,44 @@ for these associative arrays.
   *Structs* contain a fixed set of named fields, but these fields can be of any type.
   As the name suggests, these map to ``structs`` in C++.
 
-  In short, if the *key values* vary, use a dictionary.  If the *field types* vary, use a struct.
-
-Retrieving User-Defined Types from an Input File
-------------------------------------------------
+Accessing
+---------
 
-In order to have Inlet extract from the input file into a struct, it needs to know how that struct can be built.
-This is accomplished by a specializing of the struct ``FromInlet<T>`` for your type ``T``.  It must define a
-single member function with the signature ``T operator()(const inlet::Table&)``.  This function should return
-an instance of type ``T`` with its members populated with the corresponding fields in the input file.
-For the simple ``Mesh`` example whose schema is defined above, the specialization might look like:
-
-.. literalinclude:: ../../examples/user_defined_type.cpp
-   :start-after: _inlet_userdef_simple_frominlet_start
-   :end-before: _inlet_userdef_simple_frominlet_end
-   :language: C++
+As with the schema definition, the ``FromInlet`` specialization for a user-defined type will work for both
+single instances of the type and arrays of the type.
 
-The ``Table::operator[]`` is used to extract fields from the input file and is automatically converted to
-the correct member variable types when the function's return value is constructed.  This conversion does
-not happen automatically for user-defined types.  If a ``Mesh`` object as defined above is located at the
-path "mesh" within the input file, it can be retrieved as follows:
+To retrieve an array of structs as a contiguous array of user-defined type, use ``std::vector``:
 
 .. code-block:: C++
 
-  Mesh mesh = inlet["mesh"].get<Mesh>();
+  auto ops = base["operators"].get<std::vector<Operator>>();
 
-As with the schema definition, the ``FromInlet`` specialization for a user-defined type will work for both
-single instances of the type and arrays of the type.
-
-Arrays in Inlet are implemented as ``std::unordered_map<int, T>``, so retrieving an array of user-defined
-types is as follows, in this case for an array of the ``BoundaryCondition`` struct:
+Some input file languages support non-contiguous array indexing, so you can also retrieve arrays 
+as ``std::unordered_map<int, T>``:
 
 .. code-block:: C++
 
-  auto mesh = inlet["bcs"].get<std::unordered_map<int, BoundaryCondition>>();
+  auto ops = inlet["operators"].get<std::unordered_map<int, Operator>>();
+
+.. note::
+  If a non-contiguous array is retrieved as a (contiguous) ``std::vector``, the elements will be
+  ordered by increasing index.
```

Comment:
It would be helpful to provide a slightly longer example for arrays/dictionaries from input (above) to schema reading to usage.